distributed/protocol/serialize.py

```diff
@@ -86,19 +87,29 @@ def buffer_callback(f):
     return header, frames
 
 
-def pickle_loads(header, frames):
-    x, buffers = frames[0], frames[1:]
-
-    writeable = header.get("writeable")
-    if not writeable:
-        writeable = len(buffers) * (None,)
+def pickle_loads(
+    header: dict[str, Any], frames: list[bytes | bytearray | memoryview | PickleBuffer]
+) -> Any:
+    pik, buffers = frames[0], frames[1:]
+
+    def ensure_writeable_flag(mv: memoryview, w: bool) -> memoryview:
+        if w and mv.readonly:
+            # Can't avoid a deep copy
+            # Note: this doesn't actually happen in real life, as both the comms
+            # layer and the SpillBuffer store everything in writeable bytearrays
+            return memoryview(bytearray(mv))
+        elif not w and not mv.readonly:
+            # Zero copy - this is just a flag
+            return mv.toreadonly()
```

Comment:
This is the only functional change (replacing `memoryview(bytes(mv))`); the rest is cosmetic